contracts/src/v0.8/automation/upkeeps/LinkAvailableBalanceMonitor.sol

```diff
@@ -216,37 +243,50 @@ contract LinkAvailableBalanceMonitor is ConfirmedOwner, Pausable, AutomationComp
   /// @param payee the address to pay
   function withdraw(uint256 amount, address payable payee) external onlyOwner {
     require(payee != address(0), "LinkAvailableBalanceMonitor: invalid payee address");
-    LINK_TOKEN.transfer(payee, amount);
+    i_linkToken.transfer(payee, amount);
     emit FundsWithdrawn(amount, payee);
   }
 
-  /// @notice Sets the top up amount
-  function setTopUpAmount(uint256 topUpAmount) external onlyOwner returns (uint256) {
-    require(topUpAmount > 0, "LinkAvailableBalanceMonitor: invalid linkTokenAddress");
-    return s_topUpAmount = topUpAmount;
+  /// @notice Sets the minimum balance for the given target address
+  function setMinBalance(address target, uint96 minBalance) external onlyOwner {
+    require(target != address(0), "LinkAvailableBalanceMonitor: invalid target address");
+    require(minBalance > 0, "LinkAvailableBalanceMonitor: invalid minBalance");
+    if (!s_targets[target].isActive) {
+      revert InvalidWatchList();
+    }
+    uint256 oldBalance = s_targets[target].minBalance;
+    s_targets[target].minBalance = minBalance;
+    emit BalanceUpdated(target, oldBalance, minBalance);
   }
 
   /// @notice Sets the minimum balance for the given target address
-  function setMinBalance(address target, uint256 minBalance) external onlyOwner returns (uint256) {
-    require(minBalance > 0, "LinkAvailableBalanceMonitor: invalid minBalance");
-    (bool exists, uint256 prevMinBalance) = s_watchList.tryGet(target);
-    if (!exists) {
+  function setTopUpAmount(address target, uint96 topUpAmount) external onlyOwner {
+    require(target != address(0), "LinkAvailableBalanceMonitor: invalid target address");
+    require(topUpAmount > 0, "LinkAvailableBalanceMonitor: invalid topUpAmount");
+    if (!s_targets[target].isActive) {
       revert InvalidWatchList();
     }
-    s_watchList.set(target, minBalance);
-    return prevMinBalance;
+    uint256 oldTopUpAmount = s_targets[target].topUpAmount;
+    s_targets[target].topUpAmount = topUpAmount;
+    emit BalanceUpdated(target, oldTopUpAmount, topUpAmount);
   }
 
   /// @notice Update s_maxPerform
-  function setMaxPerform(uint16 maxPerform) external onlyOwner {
-    emit MaxPerformUpdated(s_maxPerform, maxPerform);
+  function setMaxPerform(uint16 maxPerform) public onlyOwner {
     s_maxPerform = maxPerform;
+    emit MaxPerformUpdated(s_maxPerform, maxPerform);
```

Comment:
total nitpick, but I've lengthy debates in the past about using `*Updated` vs `*Set` and I think the consensus is - `*Updated` should be used if the variable has changed as a result up the tx. `*Set` should be used if you don't check old vs new values.

I don't care if you don't change. Just calling out.

Reply:
Perfect, again thanks for educating me. :)